# Challenge: What test to use
Using selected questions from the 2012 and 2014 editions of the European Social Survey (http://www.europeansocialsurvey.org/data/round-index.html), address the following questions. Keep track of your code and results in a Jupyter notebook or other source that you can share with your mentor. For each question, explain why you chose the approach you did.

Here is the data file (https://raw.githubusercontent.com/Thinkful-Ed/data-201-resources/master/ESS_practice_data/ESSdata_Thinkful.csv). And here is the codebook (https://thinkful-ed.github.io/data-201-resources/ESS_practice_data/ESS_codebook.html), with information about the variable coding and content.

In this dataset, the same participants answered questions in 2012 and again 2014.

 1. Did people become less trusting from 2012 to 2014? Compute results for each country in the sample.
 2. Did people become happier from 2012 to 2014? Compute results for each country in the sample.
 3. Who reported watching more TV in 2012, men or women?
 4. Who was more likely to believe people were fair in 2012, people living with a partner or people living alone?
 5. Pick three or four of the countries in the sample and compare how often people met socially in 2014. Are there differences, and if so, which countries stand out?
 6. Pick three or four of the countries in the sample and compare how often people took part in social activities, relative to others their age, in 2014. Are there differences, and if so, which countries stand out?
Submit a link to your work below.



In [1]:
import pandas as pd

data = pd.read_csv('https://raw.githubusercontent.com/Thinkful-Ed/data-201-resources/master/ESS_practice_data/ESSdata_Thinkful.csv')
data.head()

cntry  idno  year  tvtot  ppltrst  pplfair  pplhlp  happy  sclmeet  sclact  \
0    CH   5.0     6    3.0      3.0     10.0     5.0    8.0      5.0     4.0   
1    CH  25.0     6    6.0      5.0      7.0     5.0    9.0      3.0     2.0   
2    CH  26.0     6    1.0      8.0      8.0     8.0    7.0      6.0     3.0   
3    CH  28.0     6    4.0      6.0      6.0     7.0   10.0      6.0     2.0   
4    CH  29.0     6    5.0      6.0      7.0     5.0    8.0      7.0     2.0   

   gndr  agea  partner  
0   2.0  60.0      1.0  
1   2.0  59.0      1.0  
2   1.0  24.0      2.0  
3   2.0  64.0      1.0  
4   2.0  55.0      1.0

In [2]:
data.shape

(8594, 13)

In [3]:
data_year = data.groupby(['year'])
data_year.describe()

idno                                                           \
       count          mean           std  min     25%     50%     75%   
year                                                                    
6     4297.0  39549.380033  626762.41107  1.0  1052.0  1714.0  2745.0   
7     4297.0  39549.380033  626762.41107  1.0  1052.0  1714.0  2745.0   

                   tvtot            ...  agea        partner            \
             max   count      mean  ...   75%    max   count      mean   
year                                ...                                  
6     11001431.0  4291.0  3.923095  ...  62.0  103.0  4290.0  1.376690   
7     11001431.0  4295.0  3.800931  ...  62.0  114.0  4287.0  1.393049   

                                         
           std  min  25%  50%  75%  max  
year                                     
6     0.484613  1.0  1.0  1.0  2.0  2.0  
7     0.488484  1.0  1.0  1.0  2.0  2.0  

[2 rows x 88 columns]

In [4]:
df_2012 = data[data['year'] == 6]
df_2014 = data[data['year'] == 7]

In [5]:
cntry_2012 = df_2012.groupby(['cntry'])
cntry_2014 = df_2014.groupby(['cntry'])

In [6]:
cntry_2012.describe()

idno                                                                  \
        count          mean          std         min          25%         50%   
cntry                                                                           
CH      773.0  1.480680e+03   837.085069         5.0       761.00      1432.0   
CZ      658.0  1.445666e+03   368.654806      1001.0      1169.25      1333.5   
DE       14.0  1.100048e+07   506.292818  11000031.0  11000101.00  11000271.0   
ES     1213.0  1.467607e+03   818.301231         1.0       807.00      1456.0   
NO      731.0  1.430671e+04  4672.042957      6312.0     10347.00     14346.0   
SE      908.0  1.765173e+03  1098.083963         7.0       788.50      1585.0   

                                  year       ...   agea        partner  \
               75%         max   count mean  ...    75%    max   count   
cntry                                        ...                         
CH         2243.00      2905.0   773.0  6.0  ...  61.00   94.0   773.0   
CZ         1500.75      2151.0   658.0  6.0  ...  60.00   91.0   656.0   
DE     11000833.50  11001431.0    14.0  6.0  ...  61.25   75.0    14.0   
ES         2175.00      2865.0  1213.0  6.0  ...  61.00  103.0  1210.0   
NO        18558.00     22428.0   731.0  6.0  ...  60.00   93.0   729.0   
SE         2733.75      3746.0   908.0  6.0  ...  65.00   96.0   908.0   

                                                    
           mean       std  min  25%  50%  75%  max  
cntry                                               
CH     1.349288  0.477054  1.0  1.0  1.0  2.0  2.0  
CZ     1.432927  0.495859  1.0  1.0  1.0  2.0  2.0  
DE     1.357143  0.497245  1.0  1.0  1.0  2.0  2.0  
ES     1.372727  0.483730  1.0  1.0  1.0  2.0  2.0  
NO     1.348422  0.476798  1.0  1.0  1.0  2.0  2.0  
SE     1.387665  0.487486  1.0  1.0  1.0  2.0  2.0  

[6 rows x 96 columns]

In [7]:
cntry_2014.describe()

idno                                                                  \
        count          mean          std         min          25%         50%   
cntry                                                                           
CH      773.0  1.480680e+03   837.085069         5.0       761.00      1432.0   
CZ      658.0  1.445666e+03   368.654806      1001.0      1169.25      1333.5   
DE       14.0  1.100048e+07   506.292818  11000031.0  11000101.00  11000271.0   
ES     1213.0  1.467607e+03   818.301231         1.0       807.00      1456.0   
NO      731.0  1.430671e+04  4672.042957      6312.0     10347.00     14346.0   
SE      908.0  1.765173e+03  1098.083963         7.0       788.50      1585.0   

                                  year       ...  agea        partner  \
               75%         max   count mean  ...   75%    max   count   
cntry                                        ...                        
CH         2243.00      2905.0   773.0  7.0  ...  60.0   92.0   773.0   
CZ         1500.75      2151.0   658.0  7.0  ...  58.0   87.0   652.0   
DE     11000833.50  11001431.0    14.0  7.0  ...  66.5   89.0    14.0   
ES         2175.00      2865.0  1213.0  7.0  ...  61.0  101.0  1210.0   
NO        18558.00     22428.0   731.0  7.0  ...  61.0   94.0   730.0   
SE         2733.75      3746.0   908.0  7.0  ...  67.0  114.0   908.0   

                                                    
           mean       std  min  25%  50%  75%  max  
cntry                                               
CH     1.376455  0.484810  1.0  1.0  1.0  2.0  2.0  
CZ     1.420245  0.493977  1.0  1.0  1.0  2.0  2.0  
DE     1.142857  0.363137  1.0  1.0  1.0  1.0  2.0  
ES     1.398347  0.489760  1.0  1.0  1.0  2.0  2.0  
NO     1.390411  0.488177  1.0  1.0  1.0  2.0  2.0  
SE     1.386564  0.487231  1.0  1.0  1.0  2.0  2.0  

[6 rows x 96 columns]

### Did people become less trusting from 2012 to 2014? Compute results for each country in the sample.

In [8]:
cntry_2012 = df_2012.groupby('cntry')
cntry_2014 = df_2014.groupby('cntry')

trst_2012 = cntry_2012['ppltrst'].mean()
trst_2014 = cntry_2014['ppltrst'].mean()

print(trst_2014 - trst_2012)

cntry
CH    0.073739
CZ    0.062138
DE    0.142857
ES   -0.219464
NO   -0.050685
SE    0.199210
Name: ppltrst, dtype: float64


**Generally speaking, people seemed to become more trusting from 2012 to 2014. Sweden showed the greatest increase in trust from 2012 to 2014 (0.199210), followed by Denmark (0.142857), Switzerland (0.073739), and the Czech Republic (0.062138). The two countries whose trust scores decreased were Norway (-0.050685) and Spain (-0.199210).**

### Did people become happier from 2012 to 2014? Compute results for each country in the sample.

In [9]:
happy_2012 = cntry_2012['happy'].mean()
happy_2014 = cntry_2014['happy'].mean()

print(happy_2014 - happy_2012)

cntry
CH    0.028118
CZ    0.143213
DE    0.428571
ES   -0.128713
NO   -0.336535
SE    0.039574
Name: happy, dtype: float64


**Again, it appears that, generally speaking, as a whole people felt happier from 2012 to 2014. Denmark showed the greatest increase (0.428571), followed by the Czech Republic (0.143213), Sweden (0.039574), and Switzerland (0.028118). Spain (-0.128713) and Norway (-0.336535) both decreased.**

### Who reported watching more TV in 2012, men or women?

In [10]:
men = df_2012[df_2012['gndr'] == 1.0]
women = df_2012[df_2012['gndr'] == 2.0]

In [11]:
men.describe()

idno    year        tvtot      ppltrst      pplfair  \
count  2.154000e+03  2154.0  2151.000000  2152.000000  2145.000000   
mean   3.447705e+04     6.0     3.901906     5.629182     5.947786   
std    5.797237e+05     0.0     1.983690     2.206040     2.128799   
min    1.000000e+01     6.0     0.000000     0.000000     0.000000   
25%    1.057250e+03     6.0     2.000000     4.000000     5.000000   
50%    1.800000e+03     6.0     4.000000     6.000000     6.000000   
75%    2.799750e+03     6.0     5.000000     7.000000     8.000000   
max    1.100140e+07     6.0     7.000000    10.000000    10.000000   

            pplhlp        happy      sclmeet       sclact    gndr  \
count  2146.000000  2145.000000  2151.000000  2132.000000  2154.0   
mean      5.291240     7.744522     5.228266     2.779550     1.0   
std       2.173261     1.722329     1.466672     0.917116     0.0   
min       0.000000     0.000000     1.000000     1.000000     1.0   
25%       4.000000     7.000000     4.000000     2.000000     1.0   
50%       5.000000     8.000000     6.000000     3.000000     1.0   
75%       7.000000     9.000000     6.000000     3.000000     1.0   
max      10.000000    10.000000     7.000000     5.000000     1.0   

              agea      partner  
count  2089.000000  2152.000000  
mean     46.965055     1.353625  
std      17.969622     0.478205  
min      15.000000     1.000000  
25%      33.000000     1.000000  
50%      46.000000     1.000000  
75%      61.000000     2.000000  
max     103.000000     2.000000

In [12]:
women.describe()

idno    year        tvtot      ppltrst      pplfair  \
count  2.143000e+03  2143.0  2140.000000  2135.000000  2132.000000   
mean   4.464775e+04     6.0     3.944393     5.496487     6.048311   
std    6.708246e+05     0.0     2.049457     2.263740     2.142947   
min    1.000000e+00     6.0     0.000000     0.000000     0.000000   
25%    1.048500e+03     6.0     2.000000     4.000000     5.000000   
50%    1.638000e+03     6.0     4.000000     6.000000     6.000000   
75%    2.688500e+03     6.0     6.000000     7.000000     8.000000   
max    1.100143e+07     6.0     7.000000    10.000000    10.000000   

            pplhlp        happy      sclmeet       sclact    gndr  \
count  2136.000000  2131.000000  2137.000000  2119.000000  2143.0   
mean      5.486891     7.702487     5.226954     2.700802     2.0   
std       2.165044     1.806792     1.491381     0.911657     0.0   
min       0.000000     0.000000     1.000000     1.000000     2.0   
25%       4.000000     7.000000     4.000000     2.000000     2.0   
50%       6.000000     8.000000     6.000000     3.000000     2.0   
75%       7.000000     9.000000     6.000000     3.000000     2.0   
max      10.000000    10.000000     7.000000     5.000000     2.0   

              agea      partner  
count  2082.000000  2138.000000  
mean     48.086455     1.399906  
std      18.407926     0.489993  
min      15.000000     1.000000  
25%      34.000000     1.000000  
50%      48.000000     1.000000  
75%      62.000000     2.000000  
max      97.000000     2.000000

**The answer to this question depends largely on which metric you choose to use. If speaking about total hours of television watched, then men watched more (2151 hours vs 2140 for women). However, if speaking about the mean number of hours per gender, then the women watched more (3.944 mean hours vs. 3.902 mean hours for men).**

### Who was more likely to believe people were fair in 2012, people living with a partner or people living alone?

In [13]:
partner_2012 = df_2012[df_2012['partner'] == 1]
alone_2012 = df_2012[df_2012['partner'] == 2]

In [14]:
partner_2012['pplfair'].mean()

6.080736012016523

In [15]:
alone_2012['pplfair'].mean()

5.856965174129353

**Based on average (mean), people with partners were more likely to believe people were fair (6.0807) compared with people living alone (5.8570).**

### Pick three or four of the countries in the sample and compare how often people met socially in 2014. Are there differences, and if so, which countries stand out?

In [16]:
sclmeet_2014 = cntry_2014['sclmeet'].mean()
sclmeet_2014

cntry
CH    5.160622
CZ    4.445802
DE    4.428571
ES    5.260116
NO    5.302326
SE    5.426211
Name: sclmeet, dtype: float64

**People in Sweden were the most likely to meet socially in 2014 (5.426211), followed by Norway (5.302326), Spain (5.260116), and Switzerland (5.160622). Scoring significantly lower were the Czech Republic (4.445802) and Denmark (4.428571).**

### Pick three or four of the countries in the sample and compare how often people took part in social activities, relative to others their age, in 2014. Are there differences, and if so, which countries stand out? Submit a link to your work below.

In [17]:
sclact_2014 = cntry_2014['sclact'].mean()
sclact_2014

cntry
CH    2.781699
CZ    2.703077
DE    2.714286
ES    2.616878
NO    2.859097
SE    2.879425
Name: sclact, dtype: float64

**These results seem to be much closer than any of the other results. Sweden and Norway had the highest scores (2.879425 and 2.859097, respectively). Switzerland (2.781699), Denmark (2.714286), Czech Republic (2.703077), and Spain (2.616878) round out the rest of the countries. These differences seem relatively unremarkable, as the range of mean scores is a mere 0.262547 out of a 10 point scale (or less than a 3% difference between the best and worst scoring countries).**